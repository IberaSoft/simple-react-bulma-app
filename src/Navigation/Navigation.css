.navigation-horizontal>* {
    display: inline-block;
}

.navigation-vertical>* {
    display: block;
}
