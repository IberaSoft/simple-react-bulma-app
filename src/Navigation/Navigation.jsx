import React from 'react';
import Link from './../Link/Link';

const Navigation = (props) => (
    <nav className="navbar-menu">
        <div className={props.className}>
            <Link isActive={props.activeIndex === 0} onClick={()=>props.changeIndex(0)}>Item 0</Link>
            <Link isActive={props.activeIndex === 1} onClick={()=>props.changeIndex(1)}>Item 1</Link>
            <Link isActive={props.activeIndex === 2} onClick={()=>props.changeIndex(2)}>Item 2</Link>
            <Link isActive={props.activeIndex === 3} onClick={()=>props.changeIndex(3)}>Item 3</Link>
            <Link isActive={props.activeIndex === 4} onClick={()=>props.changeIndex(4)}>Item 4</Link>
        </div>
    </nav>
)

export default Navigation;