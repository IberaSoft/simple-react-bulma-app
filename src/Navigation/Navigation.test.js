import React from 'react'
import Navigation from './Navigation';
import ShallowRenderer from 'react-test-renderer/shallow';

const defaultProps = [
    { title: 'Tabs' },
    { title: 'Accordion' },
    { title: 'MyIp' },
    { title: 'Menu 4' },
]

describe('Navigation component', () => {
    it('Should match the snapshot', () => {
        const renderer = new ShallowRenderer();
        const result = renderer.render(
            <Navigation
                menuItems={defaultProps}
                className="navbar-end"
                activeIndex={0}
                changeIndex={0}
                 />);
        expect(result).toMatchSnapshot();
    });
})