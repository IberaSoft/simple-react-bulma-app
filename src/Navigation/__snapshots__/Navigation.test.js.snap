// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Should match the snapshot 1`] = `
<nav
  className="navbar-menu"
>
  <div
    className="navbar-end"
  >
    <Link
      isActive={true}
      onClick={[Function]}
    >
      Tabs
    </Link>
    <Link
      isActive={false}
      onClick={[Function]}
    >
      Accordion
    </Link>
    <Link
      isActive={false}
      onClick={[Function]}
    >
      MyIp
    </Link>
    <Link
      isActive={false}
      onClick={[Function]}
    >
      Menu 4
    </Link>
  </div>
</nav>
`;
